kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowId.kt

```diff
@@ -46,9 +45,14 @@ internal constructor(
 internal fun <W : Workflow<I, O, R>, I, O : Any, R>
     W.id(key: String = ""): WorkflowId<I, O, R> = WorkflowId(this, key)
 
+private val KClass<*>.jClass: Class<*>
+  get() = this::class.java.getMethod("getJClass").invoke(this) as Class<*>
+
 internal fun WorkflowId<*, *, *>.toByteString(): ByteString = Buffer()
     .also { sink ->
-      sink.writeUtf8WithLength(type.jvmName)
+      // Unfortunately, we can't use type.java.name. There might be issue with inner class names
+      // and inlined types. Instead, we run the same code as Kotlin reflect would do.
+      sink.writeUtf8WithLength(type.jClass.name)
```

Comment:
That's my only concern.

Reply:
What is the potential issue?

Reply:
There was an issue in a unit test where type was `com.squareup.workflow.internal.WorkflowNodeTest$snapshots non-empty with children$$inlined$stateful$1`. But this happened because I used `type::java.name` instead of `type.java.name`. It's fixed now.